modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -743,6 +789,5 @@ func (h *HTTPTransport) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 
 // Interface guards
 var (
-	_ caddyfile.Unmarshaler = (*Handler)(nil)
```

Comment:
Why is this no longer a caddyfile Unmarshaler? It needs to be if it is to be used to unmarshal the Caddyfile.

Reply:
I agree it's not exactly right (because otherwise `reverse_proxy` still need to be a caddyfile.Unmarshaler), but this was done because with those changes, `httpcaddyfile.Helper` is needed to do the parse chaining.

Reply:
Thanks @francislavoie, you are right.

@mholt if you tell me how to cope with the absence of `httpcaddyfile.Helper` in `(*Handler).UnmarshalCaddyfile()` (as I need this `httpcaddyfile.Helper` instance to call `httpcaddyfile.ParseSegmentAsSubroute()` as suggested by @francislavoie in #3707), I will be happy to change this.

Reply:
I did a bit more digging in the source, and actually, I think it's okay for `reverse_proxy` to not be a `caddyfile.Unmarshaler`.

This is because it's never used as a guest module to something else. Regular directives don't strictly need to be Unmarshalers because their parsing is handled by callbacks registered with `RegisterHandlerDirective` or `RegisterDirective`, and those are passed `Helper`.

Most directives that do implement `caddyfile.Unmarshaler` only do so for the registered callbacks to call `UnmarshalCaddyfile`, but that's not actually necessary.

Reply:
It's good practice / conventional to implement that, so you _can_ unmarshal it where necessary. But true, all a directive needs is an unmarshaling _function_, not necessarily to implement an interface.

Reply:
I understand @mholt but in this case I don't think it's possible since we actually need to use `Helper` all the way down the callchain for `handle_response` to work.

Reply:
Any news on that topic?